showcase/src/elm/Utils/FileParser.elm

```diff
@@ -0,0 +1,389 @@
+        |. Parser.spaces
+        |= exposes
+
+
+
+-- literally parses the big chonk of code
+-- we don't need to manipulate imports so there's no need
+
+
+imports : Parser String
+imports =
+    Parser.chompUntil "\n\n\n"
+        |> Parser.getChompedString
+
+
+codeBlocks : Parser (List CodeBlock)
+codeBlocks =
+    let
+        importsHelper : List CodeBlock -> Parser (Step (List CodeBlock) (List CodeBlock))
+        importsHelper revImports =
+            Parser.oneOf
+                [ Parser.succeed (\imp -> Loop (imp :: revImports))
+                    |= Parser.oneOf [ parseType, parseFunction ]
+                    |. Parser.spaces
+                , Parser.succeed ()
+                    |> Parser.map (\_ -> Done (List.reverse revImports))
+                ]
+
+        parseType : Parser CodeBlock
+        parseType =
+            Parser.keyword "type"
+                |. Parser.chompUntilEndOr "\n\n\n"
+                |> Parser.getChompedString
+                |> Parser.map TypeDef
+
+        parseFunction : Parser CodeBlock
+        parseFunction =
+            (Parser.getChompedString <| Parser.chompUntil " ")
+                |> Parser.andThen
+                    (\name ->
+                        Parser.succeed (FunInfo name)
+                            |= (Parser.getChompedString <| Parser.chompUntil "\n")
+                            |. Parser.spaces
+                            |. Parser.token name
+                            |= (Parser.getChompedString <| Parser.chompUntilEndOr "\n\n\n")
+                            |> Parser.map FunDef
+                    )
+    in
+    Parser.loop [] importsHelper
+
+
+
+-----------------------
+-- HELPERS
+-----------------------
+
+
+{- parses stuff between (..) -}
+exposes : Parser (List String)
+exposes =
+    let
+        innerExposing : Parser (List String)
+        innerExposing =
+            Parser.chompUntil ")"
+                |> Parser.getChompedString
+                |> Parser.map (String.split "," >> List.map String.trim)
+    in
+    Parser.succeed identity
+        |. Parser.token "("
+        |= innerExposing
+        |. Parser.token ")"
+
+
+spacesAndComments : Parser ()
+spacesAndComments =
+    Parser.loop 0 <|
+        ifProgress <|
+            Parser.oneOf
+                [ Parser.lineComment "--"
+                , Parser.multiComment "{-" "-}" Parser.Nestable
+                , Parser.spaces
+                ]
+
+
+ifProgress : Parser a -> Int -> Parser (Step Int ())
+ifProgress parser offset =
+    Parser.succeed identity
+        |. parser
+        |= Parser.getOffset
+        |> Parser.map
+            (\newOffset ->
+                if offset == newOffset then
+                    Done ()
+
+                else
+                    Loop newOffset
+            )
+
+
+
+-- copy and pasting YuichiMorita's PR since the elm/parser library hasn't implemented this yet :((
+
+
+deadEndsToString : List DeadEnd -> String
+deadEndsToString deadEnds =
```

Comment:
```suggestion
{-| copy and pasting YuichiMorita's PR since the elm/parser library hasn't implemented this yet :((

https://github.com/elm/parser/pull/38
-}
deadEndsToString : List DeadEnd -> String
deadEndsToString deadEnds =
```